File: src/main/java/com/tencheeduard/hotelapp/entities/Review.java
package com.tencheeduard.hotelapp.entities;


import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;

@Entity
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer reviewId;

    // room also contains information about the hotel that the reviewer stayed in
    @ManyToOne
    Room room;

    @ManyToOne
    Account account;

    // out of 5
    Byte rating;

    @Lob
    @Column
    String body;

    Date date;

    Boolean isPublic;

}

File: frontend/src/app/classes/entities/Review.ts
import { Account } from "./Account"
import { Room } from "./Room"

export class Review
{
    reviewId : number

    room : Room

    account : Account

    rating : number

    body : string

    date : Date

    isPublic : boolean

    constructor(reviewId : number, room : Room, account : Account, rating : number, body : string, date : Date, isPublic : boolean)
    {
        this.reviewId = reviewId;
        this.room = room;
        this.account = account;
        this.rating = rating;
        this.body = body;
        this.date = date;
        this.isPublic = isPublic;
    }
}